enumerable.rb

```diff
@@ -0,0 +1,110 @@
+# rubocop:disable Metrics/CyclomaticComplexity
+# rubocop:disable Metrics/PerceivedComplexity
+
+module Enumerable
+  def my_each
+    return to_enum unless block_given
+
+    arr.length.times do |n|
+      yield arr[n]
+    end
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given
+
+    arr.length.times do |n, i|
+      yield(arr[i], n)
+    end
+  end
+
+  def my_select
+    return to_enum unless block_given
+
+    list = []
+
+    arr.my_each | filter |
+      if filter != 'accepted'
+        list.push(filter)
+        yield filter
+      end
+  end
+
+  def my_all?(*args)
+    return to_enum unless block_given
+
+    speech = true
+
+    if !block_given?
+      my_each { |argument| speech = false unless argument }
+    elsif !args[0].nil?
+      my_each { |argument| speech = false unless args[0] == argument }
+    else
+      my_each { |argument| speech = false unless yield argument }
+    end
+  end
+
+  def my_any?(*args)
+    return to_enum unless block_given
+
+    speech = false
+
+    if !block_given?
+      my_each { |argument| speech = true unless argument }
+    elsif !args[0].nil?
+      my_each { |argument| speech = true unless args[0] == argument }
+    else
+      my_each { |argument| speech = true unless yield argument }
+    end
+  end
```

Comment:
- [ ] Here are the following errors
<img width="670" alt="5 2" src="https://user-images.githubusercontent.com/58504655/103563335-6e969280-4ede-11eb-925e-482ae6a2f52b.PNG">

<img width="675" alt="5 1" src="https://user-images.githubusercontent.com/58504655/103563327-6ccccf00-4ede-11eb-8d30-0c9a91acbc87.PNG">